### ISSUE TITLE
Exception in subscribe callback makes `Observable.fromEvent()` drop subsequent events

### ISSUE THREAD

**Issue by vkarpov15** [Reactive-Extensions/RxJS#1356]
Simple example with rx@4.1.0 and node.js 6.9.0:

```javascript
'use strict'

const { EventEmitter } = require('events')
const { Observable } = require('rx')

const emitter = new EventEmitter()
const result$ = Observable.fromEvent(emitter, 'result')

let count = 0
result$.subscribe(() => {
  if (count++ === 1) {
    throw new Error('fail')
  }
  console.log('hi')
})

setInterval(() => {
  try {
    emitter.emit('result')
  } catch (error) {
    console.log('caught')
  }
}, 1000)
```

Output:

```
$ time node test
hi
caught
^C

real	0m9.540s
user	0m0.199s
sys	0m0.008s
$ 
```

TLDR; my `onNext` callback throws an exception, which is then caught, then the observable starts silently dropping events. Why does rx stop pulling events from the event emitter after the exception?

**Comment by paulpdaniels** [comment]
Throwing an exception is a terminal event, so the stream is finishing as soon as you throw. Internally there is an [`AutoDetachObserver`](https://github.com/Reactive-Extensions/RxJS/blob/master/src/modular/observer/autodetachobserver.js) which kills the stream if an error occurs during processing.

In general I would recommend making sure that you *never* throw in an `onNext` block, since there is no way to really handle the result and from a stream perspective it isn't terribly meaningful. The synchronous behavior you are relying on to catch the error is only present because you haven't introduced any asynchronous scheduling yet.

**Comment by vkarpov15** [comment]
Yeah I agree that not throwing in `onNext()` would be ideal, that was not expected on our end. I would expect that subsequent `emit()` calls would throw an error though...
